[CMake] Set VS startup project for PlayStation

donny-dont opened the pull request:
#### 5c169ee5f0adef64ea51ecb9c22777f415689746
<pre>
[CMake] Set VS startup project for PlayStation
<a href="https://bugs.webkit.org/show_bug.cgi?id=246967">https://bugs.webkit.org/show_bug.cgi?id=246967</a>

Reviewed by Basuke Suzuki.

In Visual Studio there is a run button that will build and run the
default project. Set the default to MiniBrowser when `ENABLE_WEBKIT` is
`ON` and jsc when its not.

* Source/JavaScriptCore/shell/PlatformPlayStation.cmake:
* Tools/MiniBrowser/playstation/CMakeLists.txt:

Canonical link: <a href="https://commits.webkit.org/255935@main">https://commits.webkit.org/255935@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eb99f279f801ec12674fb20b14f6cc10e17e5232

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/94086 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/3276 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/24626 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/103711 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/164052 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/98077 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/3293 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/31498 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86393 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/99752 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/99745 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/2362 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/80505 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/29380 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/84288 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/24626 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/72332 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/85330 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/66/builds/37885 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/24626 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/80463 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/35765 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/24626 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/27699 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4100 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/39639 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/41636 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/83122 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/41580 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/24626 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/18762 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

basuke commented on Tools/MiniBrowser/playstation/CMakeLists.txt:
```diff
@@ -27,9 +27,10 @@ WEBKIT_EXECUTABLE(MiniBrowser)
     set_target_properties(MiniBrowser PROPERTIES VS_DEBUGGER_WORKING_DIRECTORY "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}")
+
+    # Set the startup target to MiniBrowser
+    set_property(DIRECTORY ${PROJECT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT MiniBrowser)
```
What happens when ENABLE_WEBCORE is ON and both framework, JSC and MiniBrowser, set the property? Does later wins simply?

donny-dont replied:
https://github.com/WebKit/WebKit/pull/5718/files#r1003839868 answers your question

donny-dont commented on Source/JavaScriptCore/shell/PlatformPlayStation.cmake:
```diff
@@ -20,11 +20,15 @@ endif ()
-endif ()
 
+    # Set the startup target to JSC if WebCore disabled
+    if (NOT ENABLE_WEBCORE)
+        set_property(DIRECTORY ${PROJECT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT jsc)
+    endif ()
```
In this case jsc is only set as the startup project if `NOT ENABLE_WEBCORE`. It is not possible to build MiniBrowser without `ENABLE_WEBCORE` being `ON` so this means there will only ever be a single `VS_STARTUP_PROJECT` set.

basuke replied:
Ah, sorry. That's right. I miss understanding the case.

basuke reviewed (approved):
Thanks!

donny-dont committed:
[CMake] Set VS startup project for PlayStation
https://bugs.webkit.org/show_bug.cgi?id=246967

Reviewed by Basuke Suzuki.

In Visual Studio there is a run button that will build and run the
default project. Set the default to MiniBrowser when `ENABLE_WEBKIT` is
`ON` and jsc when its not.

* Source/JavaScriptCore/shell/PlatformPlayStation.cmake:
* Tools/MiniBrowser/playstation/CMakeLists.txt:

Canonical link: https://commits.webkit.org/255935@main
